src/tasks/dump_db.rs

```diff
@@ -0,0 +1,260 @@
+use std::{
+    collections::BTreeMap,
+    path::{Path, PathBuf},
+};
+
+use crate::{background_jobs::Environment, uploaders::Uploader, util::errors::std_error_no_send};
+
+use swirl::PerformError;
+
+/// Create CSV dumps of the public information in the database, wrap them in a
+/// tarball and upload to S3.
+#[swirl::background_job]
+pub fn dump_db(env: &Environment) -> Result<(), PerformError> {
+    // TODO make path configurable
+    const EXPORT_DIR_TEMPLATE: &str = "/tmp/dump-db/%Y-%m-%d-%H%M%S";
+    let export_dir = PathBuf::from(chrono::Utc::now().format(EXPORT_DIR_TEMPLATE).to_string());
+    std::fs::create_dir_all(&export_dir)?;
+    let visibility_config = toml::from_str(include_str!("dump-db.toml")).unwrap();
+    let database_url = if cfg!(test) {
+        crate::env("TEST_DATABASE_URL")
+    } else {
+        crate::env("DATABASE_URL")
+    };
+    run_psql(&visibility_config, &database_url, &export_dir)?;
+    let tarball = create_tarball(&export_dir)?;
+    let target_name = format!("dumps/{}", tarball.file_name().unwrap().to_str().unwrap());
+    upload_tarball(&tarball, &target_name, &env.uploader)?;
+    // TODO: more robust cleanup
```

Comment:
If an error occurs in the middle of the process, we leak the tarball and the export directory, so we will fill up `/tmp` over time if there are persistent errors with the dump. I was thinking of using a wrapper with a `Drop` implementation for the cleanup, so it will be called even in the case of errors.